capstoneteamb/capstone360
# Close Reviews

New Features:
* 'Close Reviews' button on Professor Dashboard
-How to use it:
---Go to Prof Dashboard
---Go to Edit
---Click 'Close Reviews'
-How to Test It:
----Are you able to click it?
----Does it take you to /setAvailable/?

* Student 'active' attribute has a new use as detecting open/closed reviews
-How to use it:
---create new students (mock db won't have the correct initial value)
---use all methods of adding students -- dashboard, csv, student self-adding
-How to test it:
---are the new students given 'open' for active?

* Professor can close/open reviews
-How to use it:
-----As described above, go to /setAvailable/
-----Use the dropdown to select open or closed
-How to test it:
-----Test the following cases. When set to 'closed' ensure no active reviews. When set to 'open' ensure active reviews (be mindful of any mock data which may not have been initialized correctly, you can do an initial set to 'open' to fix this per session)

Review Type, Open/Closed, Reviews should be Available/Not Available, marked as late?
on time midterm - fresh, open, available, no
on time midterm - edit on time, open, available, no
on time midterm - edit late, open, available, no
late midterm - edit late, open, available, yes


Then repeat for final [you should not need to repeat all edge cases for final, just ensure that it works here too]

As a final check, put a student in two different capstone sessions. Allow them to access reviews in one and not the other. Are they blocked from the correct review? If you switch the accessible and inaccessible reviews, are they still blocked from the correct review?

Also test: if you open up the review, then switch reviews to be closed, then try to submit the review, are you able to? (you should get a message saying they are closed)

Bug Fixes:
* Error when there is no matching professor for a capstone session
-How to use it:
----have capstone sessions in the database with an assigned prof id that does not match an existing prof id
-how to test it:
----do those capstones show up? (they shouldn't in this set up, that may be something to change later)
----do you experience any crashes?

*Changed header on set date to 'set date' from 'add team'

***Remember to check all application features!***
